### ISSUE TITLE
[ide] find references doesn't work on static fields when find_descendants/find_base is true

### ISSUE THREAD

**Issue by nadako** [HaxeFoundation/haxe#9440]

**Comment by nadako** [comment]
@RealyUniqueName looks like it's broken for static fields in general, might be related to the find-base-and-children change, could you look into it?

**Comment by nadako** [comment]
Yeah, it's because of:

https://github.com/HaxeFoundation/haxe/blob/97401b65570a5f0cab3571f62dc19e5592e744b4/src/context/display/findReferences.ml#L66

It should completely ignore these flags when we're talking about static fields.

**Comment by nadako** [comment]
Not technically a regression, but since vshaxe now always passes the flag, it stopped working in newer haxe :)

**Comment by nadako** [comment]
Should be trivial to fix after #9442 is merged.

**Comment by RealyUniqueName** [comment]
I'll cherry-pick this and #9442 when I get to updating 4.1_bugfix

**Comment by RealyUniqueName** [comment]
merged into 4.1_bugfix
